plugin.rb

```diff
@@ -63,18 +65,36 @@ def vote_count
       end
 
       def user_voted
-        if scope.user
-          object.topic.user_voted(scope.user)
-        else
-          false
-        end
+        scope.user ? object.topic.user_voted?(scope.user) : false
       end
     end
 
+    TopicQuery.results_filter_callbacks << ->(_type, result, user, options) {
+      result = result.includes(:vote_counter)
+      result = result.select("*, (SELECT COUNT(*) AS current_user_voted FROM discourse_voting_votes WHERE user_id = #{user.id} AND topic_id = topics.id)") if user
+      result
+    }
+
+    TopicQuery.results_filter_callbacks << ->(_type, result, _user, options) {
+      return result if options[:order] != "votes"
+      sort_dir = (options[:ascending] == "true") ? "ASC" : "DESC"
+      result
+        .joins("LEFT JOIN discourse_voting_vote_counters ON discourse_voting_vote_counters.topic_id = topics.id")
+        .reorder("COALESCE(discourse_voting_vote_counters.counter,'0')::integer #{sort_dir}")
+    }
+
+    TopicQuery.results_filter_callbacks << ->(_type, result, user, options) {
+      return result if options[:order] != "my_votes" || !user
```

Comment:
`order:my_votes` seems odd because it is not doing just ordering. Maybe we should add an advanced filter so that it is `in:my_votes order:votes`?